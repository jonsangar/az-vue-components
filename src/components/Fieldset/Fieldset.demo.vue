<template lang="pug">
  .inputDemo
    Demo(title="Uso Básico")
      az-fieldset(title="Datos de usuario")
        az-input(v-model="name" label="Nombre" placeholder="Nombre")
        az-input(v-model="lastName" label="Apellidos" placeholder="Apellidos")
        az-input(v-model="email" label="Email" placeholder="Email")

    Demo.no-margin-bottom(title="Slots")
      az-fieldset.no-margin-bottom
        template(v-slot:title)
          h4 Slot título
        az-input(v-model="name" label="Nombre" placeholder="Nombre")
        az-input(v-model="lastName" label="Apellidos" placeholder="Apellidos")
        az-input(v-model="email" label="Email" placeholder="Email")
</template>

<script>
export default {
  data() {
    return {
      name: '',
      lastName: '',
      email: '',
      name2: '',
      lastName2: '',
      email2: '',
    };
  },
};
</script>

<style lang="scss">
.inputDemo {
  min-height: 100vh;
  background-color: color(gray7);

  .demo__content {
    padding: 0;
  }
}
</style>
