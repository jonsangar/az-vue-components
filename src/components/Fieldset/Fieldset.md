---
device: true
---

# Fieldset

### Instalación

```javascript
import { Fieldset } from 'didor';

Vue.use(Fieldset);
```

---

### Uso

#### Uso básico

``` html
<az-fieldset title="Datos de usuario">
  <az-input v-model="name" label="Nombre" placeholder="Nombre"/>
  <az-input v-model="lastName" label="Apellidos" placeholder="Apellidos"/>
  <az-input v-model="email" label="Email" placeholder="Email"/>
</az-fieldset>
```

#### Uso de slots

``` html
<az-fieldset>
  <template v-slot:title>
    <h4>Slot título</h4>
  </template>

  <az-input v-model="name" label="Nombre" placeholder="Nombre"/>
  <az-input v-model="lastName" label="Apellidos" placeholder="Apellidos"/>
  <az-input v-model="email" label="Email" placeholder="Email"/>
</az-fieldset>
```

---

### API

| Attribute   | Description                                | Type      | Default   |
| ----------- | ------------------------------------------ | --------- | --------- |
| title | Título del grupo | `String` | -- |

### Slots

| Nombre | Descripción                                             |
| ------ | ------------------------------------------------------- |
| title | Título del grupo |
