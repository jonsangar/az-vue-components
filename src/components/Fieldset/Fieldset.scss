.az-fieldset {
  $rootClass: &;

  margin-bottom: simple();

  &--brand-theme{
    background-image: linear-gradient(to right, color(brand-dark), color(brand));
    border-radius: 10px;
  }

  .az-fieldset__content {
    border-top: none;
    border-bottom: none;
    // border-radius: 5px !important;
  }
}

@include media(palm) {
  .az-fieldset {
    &__content {
      border-top: 1px solid color(border);
      border-bottom: 1px solid color(border);
    }

    &--brand-theme{
      .az-fieldset__content {
        border-top: none;
        border-bottom: none;
        // border-radius: 5px !important;
      }
    }

    &__title {
      padding: quarter() half();
    }
  }
}
