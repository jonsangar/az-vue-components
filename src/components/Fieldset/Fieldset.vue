<template lang="pug">
  .az-fieldset
    .az-fieldset__title(v-if="title || $slots.title")
      slot(name="title")
        | {{ title }}
    .az-fieldset__content(
      :class="{'shadow-3': boxShadow}")
      slot
</template>

<script>
export default {
  props: {
    title: {
      type: String,
      default: '',
    },

    boxShadow: {
      type: Boolean,
      default: true
    }
  },
};
</script>

<style src="./Fieldset.scss" lang="scss"></style>
