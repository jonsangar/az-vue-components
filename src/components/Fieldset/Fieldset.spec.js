import { shallowMount, createLocalVue } from '@vue/test-utils';
// Importamos librerías de estilos e iconos
import SvgIcon from 'vue-svgicon'
import Fieldset from './Fieldset.vue';

const localVue = createLocalVue()
localVue.use(SvgIcon, {
  tagName: 'az-icon',
  classPrefix: 'az',
})

describe('Fieldset.vue', () => {
  const build = () => {
    const wrapper = shallowMount(Fieldset, {
      localVue,
    })

    return {
      wrapper,
    }
  }

  it('Comprueba renderizado del componente Fieldset con snapshot', () => {
    const { wrapper } = build()
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('Comprueba que la propiedad boxShadow es false por defecto y no existe la clase shadow-3', async () => {
    const { wrapper } = build();
    // Comprobamos que se tiene las props por defecto
    expect(wrapper.vm.boxShadow).toStrictEqual(false);
    expect(wrapper.vm.title).toStrictEqual('');
    // Comprobamos que se ha añadido la clase shadow-3
    expect(wrapper.find('.shadow-3').exists()).toBe(false);
  });

  it('Comprueba que la propiedad boxShadow es true y existe la clase shadow-3', async () => {
    const { wrapper } = build();
    // Si actualizamos la propiedad boxsadow a true, la clase shadow-3 existe
    await wrapper.setProps({
      boxShadow: true,
    });
    await wrapper.vm.$nextTick()
    // Comprobamos que se tiene las props por defecto
    expect(wrapper.vm.boxShadow).toStrictEqual(true);
    expect(wrapper.vm.title).toStrictEqual('');
    // Comprobamos que se ha añadido la clase shadow-3
    expect(wrapper.find('.shadow-3').exists()).toBe(true);
  });

  it('Comprueba la propiedad title con valor y que existe el texto', async () => {
    const { wrapper } = build();
    // Si actualizamos la propiedad boxsadow a false, la clase shadow-3 no existe
    await wrapper.setProps({
      title: 'myTitle',
    });
    // Comprobamos que se tiene las props por defecto
    expect(wrapper.vm.boxShadow).toStrictEqual(false);
    expect(wrapper.vm.title).toStrictEqual('myTitle');
    // Comprobamos que se ha añadido el texto myTitle
    expect(wrapper.find('.az-fieldset__title').text()).toBe('myTitle');
  });
})
