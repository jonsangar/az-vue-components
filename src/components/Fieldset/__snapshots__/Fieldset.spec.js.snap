// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Fieldset.vue Comprueba renderizado del componente Fieldset con snapshot 1`] = `
<div class="az-fieldset">
  <!---->
  <div class="az-fieldset__content"></div>
</div>
`;
